sympy/concrete/summations.py

```diff
@@ -1308,7 +1308,10 @@ def eval_sum_hyper(f, i_a_b):
                 return Piecewise(res, (old_sum, True))
         else:
             res1 = _eval_sum_hyper(f, i, a)
-            res2 = _eval_sum_hyper(f, i, b + 1)
+            try:
+                res2 = _eval_sum_hyper(f, i, b + 1)
+            except IndexError:  # for f = Trace (or other matexpr?)
+                res2 = None
```

Comment:
Where is the IndexError coming from? It doesn't seem like a reasonable exception to catch here.

Reply:
`Trace(A).rewrite(Sum)` -- the `hyper_sum` method is attempted and that method requires sampling at `upper limit + 1` which you cannot do with a physical object like a matrix.

Reply:
I don't quite follow how `hyper_sum` ends up being called. Presumably this is a case where it shouldn't be called.

Reply:
caught at source, now